include/ppx/scene/scene_gltf_loader.h

```diff
@@ -0,0 +1,396 @@
+#undef LoadImage
+#endif
+
+namespace ppx {
+namespace scene {
+
+class GltfMaterialSelector
+{
+public:
+    GltfMaterialSelector();
+    virtual ~GltfMaterialSelector();
+
+    virtual std::string DetermineMaterial(const cgltf_material* pGltfMaterial) const;
+};
+
+// -------------------------------------------------------------------------------------------------
+
+class GltfLoader
+{
+private:
+    // Make non-copyable
+    GltfLoader(const GltfLoader&)            = delete;
+    GltfLoader& operator=(const GltfLoader&) = delete;
+
+protected:
+    GltfLoader(
+        const std::filesystem::path& filePath,
+        const std::filesystem::path& textureDirPath,
+        cgltf_data*                  pGltfData,
+        bool                         ownsGtlfData,
+        scene::GltfMaterialSelector* pMaterialSelector,
+        bool                         ownsMaterialSelector);
+
+public:
+    ~GltfLoader();
+
+    // Calling code is responsbile for lifetime of pMaterialSeelctor
+    static ppx::Result Create(
+        const std::filesystem::path& filePath,
+        const std::filesystem::path& textureDirPath,
+        scene::GltfMaterialSelector* pMaterialSelector, // Use null for default selector
+        GltfLoader**                 ppLoader);
+
+    // Calling code is responsbile for lifetime of pMaterialSeelctor
+    static ppx::Result Create(
+        const std::filesystem::path& path,
+        scene::GltfMaterialSelector* pMaterialSelector, // Use null for default selector
+        GltfLoader**                 ppLoader);
+
+    cgltf_data* GetGltfData() const { return mGltfData; }
+    bool        HasGltfData() const { return (mGltfData != nullptr); }
+
+    uint32_t GetSamplerCount() const;
+    uint32_t GetImageCount() const;
+    uint32_t GetTextureCount() const;
+    uint32_t GetMaterialCount() const;
+    uint32_t GetMeshCount() const;
+    uint32_t GetNodeCount() const;
+    uint32_t GetSceneCount() const;
+
+    // These functions return a -1 if the object cannot be located by name.
+    int32_t GetSamplerIndex(const std::string& name) const;
+    int32_t GetImageIndex(const std::string& name) const;
+    int32_t GetTextureIndex(const std::string& name) const;
+    int32_t GetMaterialIndex(const std::string& name) const;
+    int32_t GetMeshIndex(const std::string& name) const;
+    int32_t GetNodeIndex(const std::string& name) const;
+    int32_t GetSceneIndex(const std::string& name) const;
+
+    // const cgltf_sampler*  FindSampler(const std::string& name) const;
+    // const cgltf_image*    FindImage(const std::string& name) const;
+    // const cgltf_texture*  FindTexture(const std::string& name) const;
+    // const cgltf_material* FindMaterial(const std::string& name) const;
+    // const cgltf_mesh*     FindMesh(const std::string& name) const;
+    // const cgltf_node*     FindNode(const std::string& name) const;
+    // const cgltf_scene*    FindScene(const std::string& name) const;
```

Comment:
@chaoticbob are these functions that we'll need in the future? (from original branch). if not, i'll remove them

Reply:
removed